pkg/operator/controllers/previewfeature/nsgflowlogs/nsgflowlogs_test.go

```diff
@@ -0,0 +1,228 @@
+					TrafficAnalyticsInterval: to.Int32Ptr(0),
+					WorkspaceID:              to.StringPtr(""),
+				},
+			},
+			ProvisioningState: "",
+		},
+		Location: to.StringPtr(location),
+	}
+}
+
+func TestReconcileManager(t *testing.T) {
+	for _, tt := range []struct {
+		name              string
+		subnetMock        func(*mock_subnet.MockManager, *mock_subnet.MockKubeManager)
+		instance          func(*aropreviewv1alpha1.PreviewFeature)
+		flowLogClientMock func(*mock_network.MockFlowLogsClient)
+		wantErr           error
+	}{
+		{
+			name: "do not enable flow log if parameters are missing/wrong",
+			subnetMock: func(mock *mock_subnet.MockManager, kmock *mock_subnet.MockKubeManager) {
+				kmock.EXPECT().List(gomock.Any()).Return([]subnet.Subnet{
+					{
+						ResourceID: resourceIdMaster,
+					},
+					{
+						ResourceID: resourceIdWorker,
+					},
+				}, nil)
+				mock.EXPECT().Get(gomock.Any(), gomock.Any()).Return(&mgmtnetwork.Subnet{
+					SubnetPropertiesFormat: &mgmtnetwork.SubnetPropertiesFormat{
+						NetworkSecurityGroup: &mgmtnetwork.SecurityGroup{
+							ID: &subnetNameMasterNSGID,
+						},
+					},
+				}, nil)
+				mock.EXPECT().Get(gomock.Any(), gomock.Any()).Return(&mgmtnetwork.Subnet{
+					SubnetPropertiesFormat: &mgmtnetwork.SubnetPropertiesFormat{
+						NetworkSecurityGroup: &mgmtnetwork.SecurityGroup{
+							ID: &subnetNameWorkerNSGID,
+						},
+					},
+				}, nil)
+			},
+			instance: func(feature *aropreviewv1alpha1.PreviewFeature) {
+				feature.Spec.NSGFlowLogs.Enabled = true
+			},
+			wantErr: fmt.Errorf("parsing failed for . Invalid resource Id format"),
+		},
+		{
+			name: "enable flow log",
+			subnetMock: func(mock *mock_subnet.MockManager, kmock *mock_subnet.MockKubeManager) {
+				kmock.EXPECT().List(gomock.Any()).Return([]subnet.Subnet{
+					{
+						ResourceID: resourceIdMaster,
+					},
+					{
+						ResourceID: resourceIdWorker,
+					},
+				}, nil)
+				mock.EXPECT().Get(gomock.Any(), gomock.Any()).Return(&mgmtnetwork.Subnet{
+					SubnetPropertiesFormat: &mgmtnetwork.SubnetPropertiesFormat{
+						NetworkSecurityGroup: &mgmtnetwork.SecurityGroup{
+							ID: &subnetNameMasterNSGID,
+						},
+					},
+				}, nil)
+				mock.EXPECT().Get(gomock.Any(), gomock.Any()).Return(&mgmtnetwork.Subnet{
+					SubnetPropertiesFormat: &mgmtnetwork.SubnetPropertiesFormat{
+						NetworkSecurityGroup: &mgmtnetwork.SecurityGroup{
+							ID: &subnetNameMasterNSGID, // this intentionally has the same NSG ID as the master subnet. master and worker subnet share the same NSG as they do in a default ARO installation
+						},
+					},
+				}, nil)
+			},
+			flowLogClientMock: func(client *mock_network.MockFlowLogsClient) {
+				flowLog := getValidFlowLogFeature()
+				flowLog.FlowLogPropertiesFormat.TargetResourceID = to.StringPtr("/subscriptions/0000000-0000-0000-0000-000000000000/resourceGroups/vnet-rg/providers/Microsoft.Network/networkSecurityGroups/masterNSG")
```

Comment:
Do we want to do something like this? 

```suggestion
				flowLog.FlowLogPropertiesFormat.TargetResourceID = &subnetNameMasterNSGID
```